src/Main.java

```diff
@@ -1,153 +1,63 @@
-
     public static void main(String[] args) {
         VehicleCollection vehicleCollection = new VehicleCollection("types", "vehicles", "rents");
         List<Vehicle> vehicleList = vehicleCollection.getVehicleList();
         MechanicService mechanicService = new MechanicService();
 
-        fixVehicles(vehicleList, mechanicService);
-
-        mechanicService.detectBreaking(vehicleList.get(0));
-        mechanicService.repair(vehicleList.get(0));
+        List<Vehicle> brokenVehicleList = vehicleCollection.getBrokenVehicles(mechanicService);
+        showVehicles(brokenVehicleList);
 
-        Rent rent = null;
-        try {
-            rent = rentVehicle(vehicleList.get(0), 100, mechanicService);
-        } catch (DefectedVehicleException e) {
-            e.printStackTrace();
-        }
+        List<Vehicle> sortedBrokenVehicleList = VehicleCollection.sortByNumberOfBrokenDetails(brokenVehicleList);
+        showVehicles(sortedBrokenVehicleList);
 
-        getRentInfo(rent);
-    }
-
-    private static class VehicleUtils {
-        private static void showVehicleArray() {
-            for (Vehicle vehicle :
-                    vehicles) {
-                System.out.println(vehicle);
-            }
-        }
-    }
+        showVolkswagenVehicles(vehicleList);
+        showNewestVolkswagen(vehicleList);
 
-    private static Rent rentVehicle(Vehicle vehicle, double cost,  MechanicService mechanicService)
-            throws DefectedVehicleException {
-        if (!mechanicService.isBroken(vehicle)) {
-            return new Rent(vehicle.getId(), new Date(), cost);
-        }
+        VehicleWash.washVehicles(vehicleList);
+        VehicleGarage.pullIntoAndLeaveGarage(vehicleList);
 
-        throw new DefectedVehicleException("Vehicle is defected");
+        repairVehicleList(vehicleList, mechanicService);
     }
 
-    private static void getRentInfo(Rent rent) {
-        if (rent != null) {
-            System.out.println(rent);
-        } else {
-            System.out.println("Rent is not possible");
-        }
-    }
-
-    private static void fixVehicles(List<Vehicle> vehicleList, MechanicService mechanicService) {
-        int max = 0;
-        Vehicle vehicleWithMaxBreaks = null;
-
-        for (int i = 0; i < vehicleList.size(); i++) {
-            Vehicle vehicle = vehicleList.get(i);
-            Map<String, Integer> map = mechanicService.detectBreaking(vehicle);
-            int breaksCount = getBreaksCount(map);
-
-            if (breaksCount == 0) {
-                System.out.println("Vehicle without breaks: " + vehicle);
-            }
-
-            if (breaksCount > max) {
-                max = breaksCount;
-                vehicleWithMaxBreaks = vehicle;
-            }
-
-            mechanicService.repair(vehicle);
-        }
-
-        System.out.println("Car with the maximum number of breaks: " + vehicleWithMaxBreaks);
+    private static void showVehicles(List<Vehicle> vehicleList) {
+        vehicleList
```

Comment:
stream is unnecessary here. You can call just forEach().

Reply:
fixed